File: CS-Examples/03_ExtractContent/FromParagraphToTable.cs
using System;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace FromParagraphToTable
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new source document and load it from a file
            Document sourceDocument = new Document();
            sourceDocument.LoadFromFile(@"..\..\..\..\..\..\Data\IncludingTable.docx");

            // Create a new destination document
            Document destinationDoc = new Document();

            // Add a section to the destination document
            Section destinationSection = destinationDoc.AddSection();

            // Extract content by table from the source document to the destination document
            ExtractByTable(sourceDocument, destinationDoc, 1, 1);

            // Save the destination document to a file named "Output.docx"
            destinationDoc.SaveToFile("Output.docx", FileFormat.Docx);

            // Dispose of the source and destination documents
            sourceDocument.Dispose();
            destinationDoc.Dispose();

            // Launch the Word file for viewing
            WordDocViewer("Output.docx");
        }

        // Extracts content by table from the source document to the destination document
        private static void ExtractByTable(Document sourceDocument, Document destinationDocument, int startPara, int tableNo)
        {
            // Get the specified table from the source document
            Table table = sourceDocument.Sections[0].Tables[tableNo - 1] as Table;

            // Get the index of the table in the source document
            int index = sourceDocument.Sections[0].Body.ChildObjects.IndexOf(table);

            // Copy each child object from the source document to the destination document
            for (int i = startPara - 1; i <= index; i++)
            {
                DocumentObject doobj = sourceDocument.Sections[0].Body.ChildObjects[i].Clone();
                destinationDocument.Sections[0].Body.ChildObjects.Add(doobj);
            }
        }
        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB-Examples/03_ExtractContent/FromParagraphToTable.vb
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace FromParagraphToTable
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Creating a new instance of the Document class for the source document.
			Dim sourceDocument As New Document()
			' Loading the source document from a file located at a specified path.
			sourceDocument.LoadFromFile("..\..\..\..\..\..\Data\IncludingTable.docx")

			' Creating a new instance of the Document class for the destination document.
			Dim destinationDoc As New Document()

			' Adding a new section to the destination document.
			Dim destinationSection As Section = destinationDoc.AddSection()

			' Calling the ExtractByTable method to extract content from the source document and add it to the destination document.
			ExtractByTable(sourceDocument, destinationDoc, 1, 1)

			' Saving the destination document to a file named "Output.docx" in the Docx file format.
			destinationDoc.SaveToFile("Output.docx", FileFormat.Docx)

			' Disposing of the resources used by the source document.
			sourceDocument.Dispose()
			' Disposing of the resources used by the destination document.
			destinationDoc.Dispose()

			'Launch the Word file.
			WordDocViewer("Output.docx")
		End Sub

		Private Sub ExtractByTable(ByVal sourceDocument As Document, ByVal destinationDocument As Document, ByVal startPara As Integer, ByVal tableNo As Integer)
			' Retrieving the table at the specified index from the first section of the source document.
			Dim table As Table = TryCast(sourceDocument.Sections(0).Tables(tableNo - 1), Table)

			' Getting the index of the table in the child objects collection of the body of the first section.
			Dim index As Integer = sourceDocument.Sections(0).Body.ChildObjects.IndexOf(table)

			' Iterating through the child objects of the body from the start paragraph to the index of the table.
			For i As Integer = startPara - 1 To index
				' Cloning the document object at the current index.
				Dim dobj As DocumentObject = sourceDocument.Sections(0).Body.ChildObjects(i).Clone()

				' Adding the cloned document object to the child objects collection of the body of the first section in the destination document.
				destinationDocument.Sections(0).Body.ChildObjects.Add(dobj)
			Next i
		End Sub
		
		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
